src/event_listener.ts

```diff
@@ -86,20 +87,37 @@ class EventListener implements EventHandler, SettingsObserver {
         this.view = view;
     }
 
-    public handleFilePathChange(path: string): void {
-        this.currentFilePath = path;
-        this.state.handleFilePathChange(path);
+    public handleFileChange(file: TFile): void {
+        this.currentFile = file;
+        this.state.handleFileChange(file);
 
     }
 
     public isCurrentFilePathIgnored(): boolean {
-        if (this.currentFilePath === null) {
+        if (this.currentFile === null) {
             return false;
         }
         const patterns = this.settings.ignoredFilePatterns.split("\n");
-        return isMatchBetweenPathAndPatterns(this.currentFilePath, patterns);
+        return isMatchBetweenPathAndPatterns(this.currentFile.path, patterns);
     }
 
+    public currentFileContainsIgnoredTag(): boolean {
+        if (this.currentFile === null) {
+            return false;
+        }
+
+        const ignoredTags = this.settings.ignoredTags.toLowerCase().split(/[\s,]+/);
+
+        const metadata = app.metadataCache.getFileCache(this.currentFile);
```

Comment:
You are now using the global app. I am not really a big fan since that can result in unexpected behavior during testing. Can you use the `this.app` from the `main.ts` and pass it downwards?

Reply:
Ahh, thanks for the tip!!